src/app/static/src/app/store/root/getters.ts

```diff
@@ -119,3 +120,20 @@ const allReviewInputsWarnings = (state: RootState) => {
     return [...sapWarnings, ...genericChartWarnings]
 }
 
+// getCalibrateOptions extracts calibrate options from Dynamic Form, this allows
+// backward compatibility supports with for rehydrating of calibrate option
+const getCalibrateOptions = (modelCalibrate: ModelCalibrateState): DynamicFormData => {
+    if (Object.keys(modelCalibrate.options).length) {
+        return modelCalibrate.options
+    }
+
+    const section = modelCalibrate.optionsFormMeta.controlSections.find(section => section.controlGroups)
```

Comment:
controlSections is an array, so by doing find I am avoid extra loop within reduce function.

Reply:
Oh, sorry, yes, it's find rather than filter. OK, so that's just going to return the first section, so you're assuming there's only ever going to be one controlSetion in the calibrate form metadata. That may not always be true though. Could just do a flatMap to get all the controlGroups for all the sections, that way you're insured against other sections getting added. 

`const allControlGroups = modelCalibrate.optionsFormMeta.controlSections.flatMap(section => section.controlGroups);`